File: SocialNetwork.Dal/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace SocialNetwork.Dal.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public DateTime Date { get; set; }
        public int AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public List<Comment> Comments { get; set; } = new List<Comment>();
        public List<Like> Likes { get; set; }
    }
}

File: SocialNetwork/ClientApp/src/app/models/post.ts
import { Like } from "./like";
import { Comment } from "./comment";
import { User } from "./users";

export class Post {
  constructor(public text: string) { }
  id: number;
  date: number;
  likes: Like[];
  comments: Comment[];
  appUserId: number;
  appUser: User;
}
